kolibri/plugins/user/assets/src/actions.js

```diff
@@ -1,58 +1,96 @@
-// const coreApp = require('kolibri');
-// const logging = require('kolibri.lib.logging');
+const PageNames = require('./state/constants').PageNames;
+const SignUpResource = require('kolibri').resources.SignUpResource;
+const coreActions = require('kolibri.coreVue.vuex.actions');
+const coreGetters = require('kolibri.coreVue.vuex.getters');
+const router = require('kolibri.coreVue.router');
 
-// const FacilityUserResource = coreApp.resources.FacilityUserResource;
-// const TaskResource = coreApp.resources.TaskResource;
-
-// const coreActions = require('kolibri.coreVue.vuex.actions');
-// const ConditionalPromise = require('kolibri.lib.conditionalPromise');
-const constants = require('./state/constants');
-const PageNames = constants.PageNames;
-// const samePageCheckGenerator = require('kolibri.coreVue.vuex.actions').samePageCheckGenerator;
-
-
-// ================================
-// USER ACTIONS
 
+function redirectToHome() {
+  window.location = '/';
+}
 
-/**
- * Actions
- *
- * These methods are used to update client-side state
- */
+function showRoot(store) {
+  const userSignedIn = coreGetters.isUserLoggedIn(store.state);
+  if (userSignedIn) {
+    router.getInstance().replace({
+      name: PageNames.PROFILE,
+    });
+    return;
+  }
+  router.getInstance().replace({
+    name: PageNames.SIGN_IN,
+  });
+}
 
 function showSignIn(store) {
+  const userSignedIn = coreGetters.isUserLoggedIn(store.state);
+  if (userSignedIn) {
+    router.getInstance().replace({
+      name: PageNames.PROFILE,
+    });
+    return;
+  }
   store.dispatch('SET_PAGE_NAME', PageNames.SIGN_IN);
   store.dispatch('SET_PAGE_STATE', {});
   store.dispatch('CORE_SET_PAGE_LOADING', false);
   store.dispatch('CORE_SET_ERROR', null);
   store.dispatch('CORE_SET_TITLE', 'User Sign In');
 }
+
+
 function showSignUp(store) {
+  const userSignedIn = coreGetters.isUserLoggedIn(store.state);
+  if (userSignedIn) {
+    router.getInstance().replace({
+      name: PageNames.PROFILE,
+    });
+    return;
+  }
   store.dispatch('SET_PAGE_NAME', PageNames.SIGN_UP);
-  store.dispatch('SET_PAGE_STATE', {});
+  store.dispatch('SET_PAGE_STATE', { signUpError: null });
```

Comment:
Usually, a `payload` object is set up there ^ for legibility. These are really small page states though, so I don't have much of an issue with it.